src/dialog-controller.ts

```diff
@@ -97,25 +99,28 @@ export class DialogController {
       return this.closePromise;
     }
 
-    return this.closePromise = invokeLifecycle(this.controller.viewModel || {}, 'canDeactivate').catch(reason => {
-      this.closePromise = undefined;
-      return Promise.reject(reason);
-    }).then(canDeactivate => {
-      if (!canDeactivate) {
-        this.closePromise = undefined; // we are done, do not block consecutive calls
-        return this.cancelOperation();
-      }
-      return this.releaseResources().then(() => {
-        if (!this.settings.rejectOnCancel || ok) {
-          this.resolve({ wasCancelled: !ok, output } as DialogCloseResult);
-        } else {
-          this.reject(createDialogCancelError(output));
-        }
-        return { wasCancelled: false };
-      }).catch(reason => {
+    const dialogResult = { wasCancelled: !ok, output } as DialogCloseResult;
+
+    return this.closePromise = invokeLifecycle(this.controller.viewModel || {}, 'canDeactivate', dialogResult)
+      .catch(reason => {
         this.closePromise = undefined;
         return Promise.reject(reason);
+      }).then(canDeactivate => {
+        if (!canDeactivate) {
```

Comment:
Ah, yes. The only breaking change concerns `.error` - always rejects with `DialogCloseError` you can see my first 2 comments.